File: 139. Word Break.py
class Solution(object):
    def wordBreak(self, s, wordDict):
        """
        :type s: str
        :type wordDict: Set[str]
        :rtype: bool
        """
        T = [False for i in xrange(len(s)+1)]
        T[0] = True
        for i in xrange(1, len(s)+1):
            for j in xrange(i):
                if T[j] and s[j:i] in wordDict:
                    T[i] = True
        return T[len(s)]

File: 139. Word Break.cpp
class Solution {
public:
    bool wordBreak(string s, unordered_set<string>& wordDict) {
        vector <bool> T (s.length()+1, false);
        T[0] = true;
        for (int i = 1; i < s.length()+1; i++)
        {
            for (int j = 0; j < i; j++)
            {
                if (T[j] && wordDict.count(s.substr(j, i - j)) > 0)
                    T[i] = true;
            }
        }
        return T[s.length()];
    }
};
